src/traveltime_google_comparison/config.py

```
@@ -111,54 +85,40 @@ def parse_args():
+        enabled = provider_data["enabled"]
+        if enabled:
+            competitor = Provider(
+                name=provider_data["name"],
+                max_rpm=int(provider_data["max-rpm"]),
+                credentials=Credentials(api_key=provider_data["api-key"]),
+            )
+            competitors.append(competitor)
+
+    if len(competitors) == 0:
         raise ValueError(
-            f"{OPENROUTES_API_KEY_VAR_NAME} not set in environment variables."
+            "There should be at least one enabled API provider that's not TravelTime"
         )
-    return openroutes_api_key
-
-
-def retrieve_tomtom_api_key():
-    tomtom_api_key = os.environ.get(TOMTOM_API_KEY_VAR_NAME)
 
-    if not tomtom_api_key:
-        raise ValueError(f"{TOMTOM_API_KEY_VAR_NAME} not set in environment variables.")
-    return tomtom_api_key
+    return Providers(base=base_provider, competitors=competitors)
 
 
-def retrieve_here_api_key():
-    here_api_key = os.environ.get(HERE_API_KEY_VAR_NAME)
-
-    if not here_api_key:
-        raise ValueError(f"{HERE_API_KEY_VAR_NAME} not set in environment variables.")
-    return here_api_key
-
-
-def retrieve_mapbox_api_key():
-    mapbox_api_key = os.environ.get(MAPBOX_API_KEY_VAR_NAME)
-
-    if not mapbox_api_key:
-        raise ValueError(f"{MAPBOX_API_KEY_VAR_NAME} not set in environment variables.")
-    return mapbox_api_key
-
-
-def retrieve_traveltime_credentials() -> TravelTimeCredentials:
-    app_id = os.environ.get(TRAVELTIME_APP_ID_VAR_NAME)
-    api_key = os.environ.get(TRAVELTIME_API_KEY_VAR_NAME)
-
-    if not (app_id and api_key):
-        raise ValueError(
-            "TravelTime API credentials are missing from environment variables."
-        )
-    return TravelTimeCredentials(app_id, api_key)
+def parse_config(file_path: str):
+    with open(file_path, "r") as file:  # letting it crash if this fails
+        content = file.read()
+        return parse_json_to_providers(content)
```

I suppose I could add some custom error handling here, but we want to let it crash anyways, so didn't seem useful